### ISSUE TITLE
feature request: use Moore's Law to calculate long time estimates

### ISSUE THREAD

**Issue by roycewilliams** [hashcat/hashcat#1553]
When requesting a long-running attack, time estimates can be quite long into the future:

```
Session..........: hashcat
Status...........: Running
Hash.Type........: scrypt
Hash.Target......: SCRYPT:1024:1:1:MDIwMzMwNTQwNDQyNQ==:5FW+zWivLxgCWj...yFcfo=
Time.Started.....: Sun Apr  1 12:53:48 2018 (3 secs)
Time.Estimated...: Fri Aug 25 23:29:29 8175 (6157 years, 146 days)
Guess.Mask.......: ?b?b?b?b?b?b?b [7]
```
(this is the example 'hashcat' scrypt hash)

It would be nice if Moore's Law could be taken into account for these time estimates. Since hashcat sessions can be saved and restored, it stands to reason that a long-running hashcat job could be moved to different hardware every 18 months or so, such that a hashcat job could run for hundreds or thousands of years. If a highly determined attacker is using this approach, the current time estimates are inaccurate.

Update: [Huang's Law](https://spectrum.ieee.org/view-from-the-valley/computing/hardware/move-over-moores-law-make-way-for-huangs-law) may be more applicable for this purpose.

**Comment by sundhaug92** [comment]
How'd one predict performance-improvements over such a long stretch? Moore's law seems to already be slowing down but we don't have that many years to extrapolate a model from.

**Comment by matlink** [comment]
Moore's law has an end. Transistors size is limited by electrons size. You could however imagine a similar law according to parallel computing grow.

**Comment by RealEnder** [comment]
For sure Time estimated have to be kept for current user's HW IMO. Maybe another line with that, but you may see coming the next request about quantums (with OpenCL support) ;)

**Comment by sundhaug92** [comment]
@RealEnder Fair point but I'm not sure QC would substantially benefit hashing-algorithms, which is usually what hashcat deals with
